packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -240,4 +255,176 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
+    
+        UNION ALL
+    
+        -- Joining Users with Comments and Votes
+        SELECT
+            public."Users"._id,
+            public."Users".username,
+            public."Users"."displayName",
+            public."Votes".power,
+            CASE
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'bigDownvote' THEN -6
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'smallDownvote' THEN -2
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'neutral' THEN 0
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'smallUpvote' THEN 2
+                WHEN public."Votes"."extendedVoteType"->>'agreement' = 'bigUpvote' THEN 6
+                ELSE 0
+            END AS agreement_value
+        FROM public."Users"
+        INNER JOIN
+            public."Comments"
+            ON public."Users"._id = public."Comments"."userId"
+        INNER JOIN
+            public."Votes"
+            ON public."Comments"._id = public."Votes"."documentId"
+        WHERE
+            public."Votes"."userId" = $1
+            AND public."Users"._id != $1
+            AND public."Votes"."votedAt" > NOW() - INTERVAL '1.5 years'
+    )
+    
+    SELECT
+        _id,
+        username,
+        "displayName",
+        SUM(power) AS total_power,
+        ARRAY_AGG(power) AS power_values,
+        COUNT(power) AS vote_counts,
+        SUM(agreement_value) AS total_agreement,
+        ARRAY(
+            SELECT val
+            FROM UNNEST(ARRAY_AGG(agreement_value)) AS val
+            WHERE val != 0
+        ) AS agreement_values
+    FROM "CombinedVotes"
+    GROUP BY _id, username, "displayName"
+    HAVING SUM(power) > 1
+    ORDER BY total_power DESC
+    LIMIT 50;
+      `, [userId])
+  }
+  
+  async getPreTopCommentersOfTopCommentedTags(topUsers: UpvotedUser[], topCommentedTags: CommentCountTag[]): Promise<UserData[]> {
+    const topUserIds = topUsers.map(user => user._id);
+    const topTagNames = topCommentedTags.map(tag => tag.name);
+  
+    // Use parameterized query to prevent SQL injection
+    const query = `
+      SELECT
+        topCommentedTags.name,
+        u.username,
+        u."displayName",
+        c."userId",
+        COUNT(*) AS post_comment_count
+      FROM unnest($1::text[]) AS topCommentedTags(name)
+      INNER JOIN public."Tags" AS t ON topCommentedTags.name = t.name
+      INNER JOIN public."TagRels" AS tr ON t._id = tr."tagId"
+      INNER JOIN public."Comments" AS c ON tr."postId" = c."postId"
+      INNER JOIN public."Users" AS u ON c."userId" = u._id
+      WHERE c."userId" = ANY($2)
+      GROUP BY topCommentedTags.name, c."userId", u.username, u."displayName"
+      HAVING COUNT(*) > 15
+    `;
+  
+    try {
+      return await this.getRawDb().any(query, [topTagNames, topUserIds]);
+    } catch (error) {
+      console.error('Error executing getAuthorsOfTopTags query:', error);
+      throw error;
+    }
```

Comment:
Why in try/catch?  (Was this for debugging?)

Reply:
We don't currently have a pattern of doing this.  Database queries generally will only throw errors if you've written it wrong.